Todo/AppDelegate.swift

```diff
@@ -0,0 +1,420 @@
+                        newRev.isDeletion = true
+                    }
+
+                    do {
+                        try newRev.saveAllowingConflict()
+                    } catch let error as NSError {
+                        NSLog("Cannot resolve conflicts with error: %@", error)
+                        return false
+                    }
+                    i++
+                }
+                return true
+            }
+    }
+
+    func nWayMergeConflicts(revs: [CBLRevision]!) ->
+        (props: [String: AnyObject]!, image: CBLAttachment?) {
+            let parent = findCommonParent(revs)
+            if parent == nil || parent!.isDeletion {
+                let defaultWinning = revs[0]
+                let props = defaultWinning.userProperties
+                let image = defaultWinning.attachmentNamed("image")
+                return (props, image)
+            }
+
+            var mergedProps = parent!.userProperties!
+            var mergedImage = parent!.attachmentNamed("image")
+            var gotTask = false, gotComplete = false, gotImage = false
+            for rev in revs {
+                if let props = rev.userProperties {
+                    if !gotTask {
+                        let task = props["task"] as? String
+                        if task != mergedProps["task"] as? String {
+                            mergedProps["task"] = task
+                            gotTask = true
+                        }
+                    }
+
+                    if !gotComplete {
+                        let complete = props["complete"] as? Bool
+                        if complete != mergedProps["complete"] as? Bool {
+                            mergedProps["complete"] = complete
+                            gotComplete = true
+                        }
+                    }
+                }
+
+                if !gotImage {
+                    let attachment = rev.attachmentNamed("image")
+                    let attachmentDiggest = attachment?.metadata["digest"] as? String
+                    if (attachmentDiggest != mergedImage?.metadata["digest"] as? String) {
+                        mergedImage = attachment
+                        gotImage = true
+                    }
+                }
+
+                if gotTask && gotComplete && gotImage {
+                    break
+                }
+            }
+            return (mergedProps, mergedImage)
+    }
+
+    func findCommonParent(revisions: [CBLRevision]!) -> CBLRevision? {
+        var minHistoryCount = 0
+        var histories : [[CBLRevision]] = []
+        for rev in revisions {
+            let history = (try? rev.getRevisionHistory()) ?? []
+            histories.append(history)
+            minHistoryCount =
+                minHistoryCount > 0 ? min(minHistoryCount, history.count) : history.count
+        }
+
+        var commonParent : CBLRevision? = nil
+        var i: Int
+        for i = 0; i < minHistoryCount; i++ {
```

Comment:
C-style `for` loops will be removed in Swift 3. They're deprecated now and Xcode 7.3 warns about them. You can either iterate over `history` itself, or use a range from 0...minHistoryCount.